### PR TITLE
[TS SDK] use `0x1::aptos_account::transfer` in tests

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
use `0x1::aptos_account::transfer` instead of `0x1::coin::transfer` and update tests.
### Test Plan
tests are passing

### CODE REVIEW

File: ecosystem/typescript/sdk/src/aptos_types/transaction.ts [hunk 1]
```diff
@@ -295,7 +295,7 @@ export class ModuleId {
   /**
    * Converts a string literal to a ModuleId
    * @param moduleId String literal in format "AccountAddress::module_name",
-   *   e.g. "0x1::coin"
+   *   e.g. "0x1::aptos_coin"
```
Comment (gregnazario) [inline ecosystem/typescript/sdk/src/aptos_types/transaction.ts:298]: this one probably could have stayed the original

File: ecosystem/typescript/sdk/src/transaction_builder/builder.ts [hunk 2]
```diff
@@ -232,7 +232,7 @@ export class TransactionBuilderABI {
    * Builds a TransactionPayload. For dApps, chain ID and account sequence numbers are only known to the wallet.
    * Instead of building a RawTransaction (requires chainID and sequenceNumber), dApps can build a TransactionPayload
    * and pass the payload to the wallet for signing and sending.
-   * @param func Fully qualified func names, e.g. 0x1::Coin::transfer
+   * @param func Fully qualified func names, e.g. 0x1::aptos_account::transfer
```
Comment (gregnazario) [inline ecosystem/typescript/sdk/src/transaction_builder/builder.ts:235]: these are good, fixes the case issue

File: ecosystem/typescript/sdk/src/transaction_builder/builder.ts [hunk 3]
```diff
@@ -380,7 +381,7 @@ export class TransactionBuilderRemoteABI {
     if (funcNameParts.length !== 3) {
       throw new Error(
         // eslint-disable-next-line max-len
-        "'func' needs to be a fully qualified function name in format <address>::<module>::<function>, e.g. 0x1::coins::transfer",
+        "'func' needs to be a fully qualified function name in format <address>::<module>::<function>, e.g. 0x1::aptos_account::transfer",
```
Comment (gregnazario) [inline ecosystem/typescript/sdk/src/transaction_builder/builder.ts:384]: also this fixes a non-existent example

**Review by gregnazario — APPROVED** [review]
cuts down on a lot of duplicate boilerplate!

**Comment by 0xjinn** [comment]
Real noob question - What is the different between coin and aptos_account and why the changes? Is this because of the update on fungible asset?

**Comment by 0xmaayan** [comment]
> Real noob question - What is the different between coin and aptos_account and why the changes? Is this because of the update on fungible asset?

`0x1::coin::transfer` assumes the receiver account exists (and therefore we needed to first create the account, i.e fund it, before transfering coins) and supports only APT coin.
`0x1::aptos_account::transfer` would create the receiver account if doesnt exist, supports APT coin
`0x1::aptos_account::transfer_coins` would create the receiver account if doesnt exist, supports any coin type

File: ecosystem/typescript/sdk/CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,10 @@ All notable changes to the Aptos Node SDK will be captured in this file. This ch
 
 ## Unreleased
 
+- export `bcsSerializeU256` from `bcs/helper.ts`
```
Comment (banool) [inline ecosystem/typescript/sdk/CHANGELOG.md:7]: nit
```suggestion
- Export `bcsSerializeU256` from `bcs/helper.ts`
```

File: ecosystem/typescript/sdk/CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,10 @@ All notable changes to the Aptos Node SDK will be captured in this file. This ch
...
+- Add `examples` folder to `.npmignore`
+- use `0x1::aptos_account::transfer` in tests
```
Comment (banool) [inline ecosystem/typescript/sdk/CHANGELOG.md:9]: nit:
```suggestion
- Use `0x1::aptos_account::transfer` in tests
```

**Review by banool — APPROVED** [review]

**Pull request merged**